OneCommunityGlobal/HighestGoodNetworkApp
# Sriram taking over for Imran toggle request bio

# Description
Added Permission that gives the user permission to toggle the "Bio announcement" switch under "Reports" -> "Weekly Summaries Reports"

## Related PRS (if any):
This frontend PR is related to the [#960](https://github.com/OneCommunityGlobal/HGNRest/pull/960) backend PR.

## Main changes explained:
- Added a new permission label for toggle request bio.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to Other Links -> Permissions Management -> Manage User Permissions -> Choose a user without Toggle Request Bio under Reports -> click on Add -> Scroll down then submit
6. verify that the user who was given the permission is now able to see the toggle on Reports -> Weekly Summaries Report and toggle successfully.



## Screenshots or videos of changes:

**As Admin/Owner Add the "Toggle Request Bio" Permission to User**

![Step 1](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/043bec1a-6a40-4a0b-8cbf-da488854887c)


**As Volunteer User Navigate to Reports -> Weekly Summaries Reports**

![Step 2](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/35c8bd47-27c1-4028-8777-d3d0a7b6ff92)


**Verify that user can interact with "Bio announcement" Switch and can Toggle its State**

![Step 3](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/93bc920f-3d73-42cf-bd1d-a94837b48339)


**Verify that if user does not have "Toggle Request Bio" Permission they are not able the button is not visible but there is a label**

![Step 4](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/d1bc1187-0cae-4585-947e-b459b32fc418)


## Note:

1. The User assigned the Permission should also have "See Weekly Summaries" permission to allow them to navigate to the Reports Page
2.  Changing the Bio announcement for the "Edwardo Mentor" returns a 404 Error  because "userProfile validation failed: warnings.3.description: `uuttt` is not a valid enum value for path `description`., warnings.4.description: `uuttt` is not a valid enum value for path `description`.". Please test for other Members.
